network/proto/senders.go

```diff
@@ -63,10 +70,37 @@ func (s defaultMessageSender) sendTransactionListQuery(peer p2p.PeerID, blockDat
 }
 
 func (s defaultMessageSender) sendTransactionList(peer p2p.PeerID, transactions []dag.Transaction, blockDate time.Time) {
-	envelope := createEnvelope()
+	if len(transactions) == 0 {
+		envelope := createEnvelope()
```

Comment:
I think we can just skip sending anything. The protocol doesn't demand a response to a transactionListQuery.